### PR TITLE
rds created, vpc created

### PR DESCRIPTION
**Pull request by DhruvJWork**
rds terraform altered
vpc created 
subnets added
using default network

### CODE REVIEW

File: rds.tf [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
+resource "aws_db_instance" "mydb1" {
+  allocated_storage        = 256 # gigabytes
+  backup_retention_period  = 7   # in days
+  db_subnet_group_name     = var.rds_public_subnet_group
+  engine                   = "postgres"
+  engine_version           = "10"
+  identifier               = "mydb1"
+  instance_class           = "db.t3.micro"
+  multi_az                 = false
+  name                     = "mydb1"
```
Comment (jennifer-gft) [inline rds.tf]: DB name must be 'gftclientdb'

File: rds.tf [hunk 2]
```diff
@@ -1,12 +1,48 @@
...
+  password                 = var.db_password
+  port                     = 5432
+  publicly_accessible      = true
+  storage_encrypted        = true # you should always do this
+  storage_type             = "gp2"
+  username                 = var.db_username
+  vpc_security_group_ids   = ["${aws_security_group.mydb1.id}"]
+}
+
+resource "aws_vpc" "main" {
```
Comment (jennifer-gft) [inline rds.tf]: vpc, subnets and other network resources are being created in main.tf, so this can be removed

File: rds.tf [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+  cidr_block           = "172.31.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+}
+
+resource "aws_security_group" "mydb1" {
+  name = "mydb1"
+
+  description = "RDS postgres servers (terraform-managed)"
+  vpc_id = "${var.rds_vpc_id}"
```
Comment (jennifer-gft) [inline rds.tf]: Must replace this h/c'd vpc id and lookup from main.tf vpc instead

File: terraform.eu-west-2.tfvars [hunk 4]
```diff
@@ -1,2 +1,4 @@
 region             = "eu-west-2"
-availability_zones = ["eu-west-2a", "eu-west-2b", "eu-west-2c"] 
\ No newline at end of file
+availability_zones = ["eu-west-2a", "eu-west-2b", "eu-west-2c"] 
+db_username         = "mydb1"
```
Comment (jennifer-gft) [inline terraform.eu-west-2.tfvars]: please change username to 'gftadmin'

File: rds.tf [hunk 5]
```diff
@@ -1,12 +1,48 @@
...
-  username              = "foo"
-  password              = "foobarbaz"
-  parameter_group_name  = "default.mysql5.7"
-  skip_final_snapshot   = true
-}
\ No newline at end of file
+resource "aws_db_instance" "mydb1" {
+  allocated_storage        = 256 # gigabytes
+  backup_retention_period  = 7   # in days
+  db_subnet_group_name     = var.rds_public_subnet_group
```
Comment (jennifer-gft) [inline rds.tf]: vpc, subnets are being created in main.tf, public subnets must be a look-up from main.tf

File: rds.tf [hunk 6]
```diff
@@ -1,12 +1,48 @@
...
+  engine                   = "postgres"
+  engine_version           = "10"
+  identifier               = "mydb1"
```
Comment (jennifer-gft) [inline rds.tf]: rename this to 'gftclientdb'

File: rds.tf [hunk 7]
```diff
@@ -1,12 +1,48 @@
...
+  instance_class           = "db.t3.micro"
```
Comment (jennifer-gft) [inline rds.tf]: Instance can be - db.t3.medium

File: variables.tf [hunk 8]
```diff
@@ -29,4 +29,32 @@ variable "availability_zones" {
   type        = list(any)
   description = "AZ of the region"
   default     = ["eu-west-2a", "eu-west-2b", "eu-west-2c"]
+}
+########
+variable "rds_vpc_id" {
```
Comment (jennifer-gft) [inline variables.tf]: rds_vpc_id, rds_public_subnets, rds_public_subnet_group can be removed

File: rds.tf (hunk 1 above)
Comment (DhruvJWork) [reply rds.tf]: resolved

File: rds.tf (hunk 2 above)
Comment (DhruvJWork) [reply rds.tf]: resolved

File: terraform.eu-west-2.tfvars (hunk 4 above)
Comment (DhruvJWork) [reply terraform.eu-west-2.tfvars]: resolved

File: rds.tf (hunk 6 above)
Comment (DhruvJWork) [reply rds.tf]: resolved

File: rds.tf (hunk 7 above)
Comment (DhruvJWork) [reply rds.tf]: resolved

File: variables.tf (hunk 8 above)
Comment (DhruvJWork) [reply variables.tf]: resolved

**Review by jennifer-gft — APPROVED** [review]

**Pull request merged**